### DISCUSSION TITLE
Rope simulation

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#4346] [Q&A]
Hello,

I was wondering how could I simulate a rope in Webot.

Use case: I want to simulate some sort of sailing boat, which indeed feature a lot of them to control the sails.

Model of a rope :  
I think that we must not try to simulate the true rope (that would be very computation intensive, if we model that like a chain of element, in term of collision / links between element...)
What I was thinking of was a SliderJoint with a max length, with BallJoint on each extremities. When it reach the max position (ie the length of the rope) it will start to pull on the extremities and apply force to get extremities at the given length. Otherwise, not force would be applied (ie the robe is not in tension). We could even add some sort of "spring" next to/with/instead of the SliderJoint to simulate the elongation of a rope.

I could see an issue with using the max position of the slider joint to monitor the rope, is that, I've been doing some time ago experiment with changing the max position during the simulation, but it's not very convincing to make the length of the rope vary during the simulation. (and that would be very handy to drive sails, for instance). Moreover I had to use the supervisor for that, and I would prefer avoiding that, as long as possible.

That my hot take on that, I'm trying to sum up my ideas, feels free to challenge it :) 

Cheers,

**Comment by ad-daniel** [comment]
We do have a model of rope in the library `rope.wbt`, but it is modeled by using several cylinders which as you said it might not scale well at all. I don't think the approach would work super well in the case of a boat, but hasn't really been tested either I think :D
As you mentioned spring, there's also `force_control.wbt` in the samples which although it doesn't represent the spring visually, it does model its behavior (but requires a supervisor/physics plugin).

As for your approach the only way to know for sure would be to try and see what happens :)

**Reply by ShuffleWire** [reply]
What is the difference between what is shown in `force_control.wbt` and using JointParameters on the join ?

Visual representation is not an issue, I could either do without, or make something representative enough to work with it.

I will continue to think about it...

**Comment by muhtesemendis** [comment]
I wanna create basic rocket simulation but I couldnt understand how can I create I couldnt find rocket simulation example

**Reply by ShuffleWire** [reply]
I don't see the relation with the OP, if you need to ask any new question please open a new thread.

But for your rocket you simply need a motor with a propeller directed to the bottom, attached to some sort of body (with a mass and maybe a drag coefficient)
